spec/Candid.md

```diff
@@ -144,10 +144,12 @@ A *service* is a standalone actor on the platform that can communicate with othe
 
 #### Structure
 
-A service's signature is described by an *actor type*, which defines the list of *methods* that the service provides. Each method is described by its *name* and a *function type* describing its signature. The function type can also be given by referring to a type definition naming a function reference type.
+A service's signature is described by an *actor type*, which defines the list of *methods* that the service provides. 
+A service can optionally provide *initialization parameters* for creating services dynamically with the same signature.
+Each method is described by its *name* and a *function type* describing its signature. The function type can also be given by referring to a type definition naming a function reference type.
 
 ```
-<actortype> ::= { <methtype>;* }
+<actortype> ::= { <methtype>;* } | <tuptype> -> { <methtype>;* }
```

Comment:
We should also allow forming
```
... | <tuptype> -> <id>
```
where `<id>` refers to an initialised actor type.

Reply:
Should we? It's symmetric with functions, which don't allow `<tuptype> -> <id>`. What does `(nat) -> service (int) -> { f: () -> () }` mean? Initializing this service gets another uninitialized service?

Reply:
Oh, I didn't suggest that you can nest them, thus _initialised_ actor type. Only that you can write this:
```
type A = service {...};
type B = service (Nat) -> A;
```
The `id` can only be a non-arrow type, i.e.,
```
type C = service (Nat) -> B;
```
would be an error.